### ISSUE TITLE
DKIM signer is always instanciated and attached, even if DKIM parameters aren't set

### ISSUE THREAD

**Issue by Dahpril** [TwakeApp/Twake#88]
**Describe the bug**
In Parameters.php, when DKIM parameters are left empty or commented, TwakeMailer service try anyway to sign mail with DKIM, throwing a fatal error.

Complete Stacktrace : 
```
Call Stack:
    0.0002     350744   1. {main}() /twake-core/bin/console:0
    0.0669    2966624   2. App\App->runCli() /twake-core/bin/console:9
    0.0669    2967080   3. Common\CommandsManager->run() /twake-core/app/App.php:134
    0.0669    2968792   4. Common\CommandsManager->execute() /twake-core/app/Common/CommandsManager.php:27
    0.0677    2985824   5. Twake\Core\Command\MailsQueueCommand->executeFromManager() /twake-core/app/Common/CommandsManager.php:46
    0.0677    2985824   6. Twake\Core\Command\MailsQueueCommand->execute() /twake-core/app/Common/Commands/ContainerAwareCommand.php: 27
   21.4116    8465976   7. Twake\Core\Services\TwakeMailer->sendInternal() /twake-core/src/Twake/Core/Command/MailsQueueCommand.php:6 6
   21.4298    9283392   8. Twake\Core\Services\TwakeMailer->sendHtml() /twake-core/src/Twake/Core/Services/TwakeMailer.php:95
   21.4834   11078552   9. Swift_Mailer->send() /twake-core/src/Twake/Core/Services/TwakeMailer.php:151
   21.8575   11103232  10. Swift_SmtpTransport->send() /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Mailer.php:71
   21.8583   11111592  11. Swift_SmtpTransport->sendTo() /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/Abstr actSmtpTransport.php:206
   21.8583   11111968  12. Swift_SmtpTransport->doMailTransaction() /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Tran sport/AbstractSmtpTransport.php:516
   21.9756   11111968  13. Swift_SmtpTransport->streamMessage() /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transpor t/AbstractSmtpTransport.php:500
   21.9761   11118448  14. Swift_Message->toByteStream() /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/Abstr actSmtpTransport.php:398
   21.9779   11122280  15. Swift_Message->doSign() /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Message.php:169
   22.0229   11125736  16. Swift_Signers_DKIMSigner->addSignature() /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Mess age.php:203
   22.0234   11128096  17. Swift_Signers_DKIMSigner->getEncryptedHash() /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/ Signers/DKIMSigner.php:551
```

**To Reproduce**
1. In Parameters.php, don't set any DKIM settings, or comment this array block
2. Try to send a mail

**Expected behavior**
Mail should be sent, even if DKIM settings aren't defined.

**Possible solution**
In TwakeMailer service (twake/backend/core/src/Twake/Core/Services/TwakeMailer.php), check if DKIM parameters are defined and are not left empty with
```php
//Line 121
if(isset($this->mail_parameters["dkim"]) && 
   isset($this->mail_parameters["dkim"]["private_key"]) && $this->mail_parameters["dkim"]["private_key"] !== "" &&
   isset($this->mail_parameters["dkim"]["domain_name"]) && $this->mail_parameters["dkim"]["domain_name"] !== "" &&
   isset($this->mail_parameters["dkim"]["selector"]) && $this->mail_parameters["dkim"]["selector"] !== ""){
    /* Instanciate Swift_Signers_DKIMSigner $signer and attach it to SwiftMailer $this->mailer */
}

// Same for attachSigner line 149
```

Maybe consider to check this in the beginning of sendHtml method and store result in variable ($is_DKIM for exemple) to avoid duplicating those complex conditions on line 121 and line 149
```php
//Line 117
$is_DKIM = false;
if(isset($this->mail_parameters["dkim"]) && 
   isset($this->mail_parameters["dkim"]["private_key"]) && $this->mail_parameters["dkim"]["private_key"] !== "" &&
   isset($this->mail_parameters["dkim"]["domain_name"]) && $this->mail_parameters["dkim"]["domain_name"] !== "" &&
   isset($this->mail_parameters["dkim"]["selector"]) && $this->mail_parameters["dkim"]["selector"] !== ""){
    $is_DKIM = true;
}

//Line 121
if($is_DKIM){
    /* Instanciate Swift_Signers_DKIMSigner $signer and attach it to SwiftMailer $this->mailer */
}

// Same for attachSigner line 149
```

**Desktop (please complete the following information):**
 - OS: Windows 10 2004
 - Browser: Firefox
 - Version: 78.0.2

Thanks ! 🤗

**Comment by RomaricMourgues** [comment]
I see you already proposed a solution!

As we would be delighted to have our first external Pull Request, do you think you can do the change directly on the file ?
The procedure is very simple:
1. Go to https://github.com/TwakeApp/Twake/blob/main/twake/backend/core/src/Twake/Core/Services/TwakeMailer.php
2. Click on the Pencil icon to edit the file
3. Once fixed, set a commit message (at the bottom of the page, use a ♻️ smiley like "♻️ Allow unconfigured DKIM in mailer")
4. Push the change to your automatically created fork

If you don't want to do this, I'll fix the error myself when I have time, but we would really appreciate a pull request coming from you :)

Thanks and have a nice day!

**Comment by RomaricMourgues** [comment]
Fixed by #90 
Thanks Dahpril 🥳

**Comment by objeni** [comment]
mail parameters give me error like following below:
The connection timed out after 5 sec while awaiting incoming data

Fatal error: Uncaught Swift_TransportException: Expected response code 250 but got code "530", with message "530 5.7.0 Must issue a STARTTLS command first. j11sm15999513pfc.64 - gsmtp
" in /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php on line 457

Swift_TransportException: Expected response code 250 but got code "530", with message "530 5.7.0 Must issue a STARTTLS command first. j11sm15999513pfc.64 - gsmtp
" in /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php on line 457


Call Stack:
    0.0001     350744   1. {main}() /twake-core/bin/console:0
    0.0785    3036200   2. App\App->runCli() /twake-core/bin/console:9
    0.0786    3036656   3. Common\CommandsManager->run() /twake-core/app/App.php:139
    0.0786    3038368   4. Common\CommandsManager->execute() /twake-core/app/Common/CommandsManager.php:27
    0.0792    3055368   5. Twake\Core\Command\MailsQueueCommand->executeFromManager() /twake-core/app/Common/CommandsManager.php:46
    0.0792    3055368   6. Twake\Core\Command\MailsQueueCommand->execute() /twake-core/app/Common/Commands/ContainerAwareCommand.php:27
    6.5418    8566384   7. Twake\Core\Services\TwakeMailer->sendInternal() /twake-core/src/Twake/Core/Command/MailsQueueCommand.php:69
    6.7314   10656536   8. Twake\Core\Services\TwakeMailer->sendHtml() /twake-core/src/Twake/Core/Services/TwakeMailer.php:100
    6.8381   12419952   9. Swift_Mailer->send() /twake-core/src/Twake/Core/Services/TwakeMailer.php:168
    7.4625   12444272  10. Swift_SmtpTransport->send() /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Mailer.php:71
    7.4633   12452632  11. Swift_SmtpTransport->sendTo() /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php:206
    7.4633   12453008  12. Swift_SmtpTransport->doMailTransaction() /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php:516
    7.4638   12454400  13. Swift_SmtpTransport->doDataCommand() /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php:497
    7.4638   12454400  14. Swift_SmtpTransport->executeCommand() /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php:390
    7.4638   12454400  15. Swift_SmtpTransport->executeCommand() /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/EsmtpTransport.php:305
    7.7151   12478704  16. Swift_SmtpTransport->throwException() /twake-core/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php:346

please Give me solution as soon as possible
